pkg/devfile/devfile.go

```diff
@@ -124,14 +124,27 @@ func GetResourceFromDevfile(log logr.Logger, devfileData data.DevfileData, deplo
 				}
 
 				if len(resources.Deployments) > 0 {
-					// update for replica
-					currentReplica := int32(component.Attributes.GetNumber(ReplicaKey, &err))
-					if err != nil {
-						if _, ok := err.(*attributes.KeyNotFoundError); !ok {
+					// need to pass in a new error holder, otherwise previous errors will persist and give a false error if key is found
+					var replicaErr error
+					currentReplica := int32(component.Attributes.GetNumber(ReplicaKey, &replicaErr))
+					if replicaErr != nil {
+						if _, ok := replicaErr.(*attributes.KeyNotFoundError); !ok {
 							return parser.KubernetesResources{}, err
+						} else {
+							// check the deployment to see what value is set
+							replica := resources.Deployments[0].Spec.Replicas
+							if replica != nil {
+								// if there is no component attribute, we will use the value in the deployment spec
+								currentReplica = *replica
+							} else {
+								// default value is 1.  We shouldn't hit this code path since this check is done in the update logic and will write the attribute deployment/replicas:1
+								currentReplica = 1
+							}
```

Comment:
I think these lines can be simplified?
1. we are checking if replica is set in deploy
2. if it is, we are reading it and setting it back 🤔 

Even when we are checking if its empty and setting it to 1, do we even need to? Its 1 by default in Kubernetes i.e. if you were to create a pod without this property there is only one repllca, no?

Reply:
It looks redundant but it's because we are searching for the component attribute key first in line 129 

`currentReplica := int32(component.Attributes.GetNumber(ReplicaKey, &replicaErr))`

We are relying on the presence of the attribute to override anything that's set in deployment but if it doesn't exist, it'll return a `KeyNotFoundError` and assign a 0 value.  Similar to the previous comment it's there to "reset" the value